web/src/components/dashboard/Sidebar.jsx

```diff
@@ -134,89 +113,40 @@ function Sidebar({ isSidebarOpen }) {
     };
   }, []);
 
-  const onClickCreateForm = async (formValues) => {
+  const addForm = async (values) => {
+    let isAddFormSuccessful = false;
+
     try {
       setIsLinearProgressModalOpen(true);
 
-      const { type, content } = formValues;
+      const { type, content } = values;
       const newForm = { type, content: {} };
 
       Object.values(content).forEach((fs) => {
         if (fs.type === FORM_COMPONENT_TYPES.IMAGE) {
           newForm.content.images = FORM_COMPONENT_TYPES.IMAGE;
         } else if (fs.name && fs.type) {
-          newForm.content[fs.name] = fs.type;
+          newForm.content[fs.name.toString()] = fs.type;
         }
       });
 
       const result = await createForm(newForm);
 
-      if (result.status === 'success') {
-        setIsFormOpen(false);
-        setSidebarItems((prev) => [{ id: type, type }, ...prev]);
-      } else {
-        toast.error(result.message);
-        return false;
-      }
-
-      return true;
-    } catch (error) {
-      console.log(error);
-      toast.error(error.message);
-      return false;
-    } finally {
-      setIsLinearProgressModalOpen(false);
-    }
-  };
-
-  const addPost = async () => {
-    try {
-      setIsLinearProgressModalOpen(true);
-      const fd = new FormData();
-      const { type, topics, content } = postCreatorMethods.getValues();
-      const { images, ...rest } = content ?? {};
-      const filledContentFields = {};
-
-      Object.entries(rest).forEach(([k, v]) => { if (v) filledContentFields[k] = v; });
-
-      if (images) {
-        (Array.isArray(images) ? images : []).forEach((image) => {
-          if (image.preview) {
-            URL.revokeObjectURL(image.preview);
-          }
-          fd.append('image', image);
-        });
-      }
-
-      fd.append('content', JSON.stringify(filledContentFields));
-      fd.append('type', JSON.stringify(type));
-      fd.append('topics', JSON.stringify(topics));
-
-      const response = await fetch(`${process.env.REACT_APP_KNOWZONE_BE_URI}/${BE_ROUTES.POSTS}`, {
-        method: 'POST',
-        body: fd,
-        credentials: 'include',
-      });
-      const result = await response.json();
-
       if (result?.status === 'fail') {
```

Comment:
update request returns updated object instead of { status: 'true', message '...' }. Thus, it is better to create branching according to fail status to have consistency.